botorch/acquisition/multi_objective/monte_carlo.py

```diff
@@ -747,7 +780,13 @@ def _get_f_X_samples(self, posterior: GPyTorchPosterior, q: int) -> Tensor:
 
         # TODO: improve efficiency for multi-task models
         samples = self.sampler(posterior)
-        return samples[..., -q:, :]
+        if isinstance(self.model, HigherOrderGP):
+            # Select the correct q-batch dimension for HOGP.
+            q_dim = -self.model._num_dimensions
+            q_idcs = torch.arange(-q, 0) + samples.shape[q_dim]
```

Comment:
We need to set the device in arange to run this on cuda

Reply:
Interesting, `gpu_tensor[cpu_idx_tensor]` works fine but `gpu_tensor.index_select(idx, cpu_idx_tensor)` doesn't.